webpack.config.js

```diff
@@ -24,7 +25,27 @@ const baseUrl = isProduction ? '' : '/'
 const config = overrideEnvConfig(loadConfig())
 const { name: appTitle } = config
 
-const apps = [SWAP_APP_V1, TRADE_APP]
+function getSelectedApps() {
+  const appName = process.env.APP
+  if (appName) {
```

Comment:
Could also avoid the Array.find with Map, though one could argue its maybe overcomplicated that way

```ts
const ALL_APPS = new Map([
    [SWAP_APP_V1.name, SWAP_APP_V1], 
    [TRADE_APP.name, TRADE_APP]
])

...

function getSelectedApps() {
  const appName = process.env.APP
  if (appName) {
    const app = ALL_APPS.get(appName)
    if (!app) {
      throw new Error(`Unknown App ${app}`)
    }

    return [
      {
        ...app,
        filename: 'index.html', // If we return only one app, the html web is "index.html"
      },
    ]
  } else {
    const allApps = Array.from(ALL_APPS.values())
    return allApps
  }
}

...

// Get selected apps: all apps by default
const apps = getSelectedApps()

Reply:
I find the find ok, I see your option also OK. but I think I like more how it reads `const ALL_APPS = [SWAP_APP_V1, TRADE_APP]`

but I might do this in the top of the waterfall since you bother to write it and is also good option